app/controllers/application_controller.rb

```diff
@@ -12,6 +13,17 @@ def authenticate_admin!
     end
   end
 
+  ##
+  # Forbid the request if the back_uri parameter is being set to create
+  # offsite links.  The parameter is used in various views throughout the
+  # application.
+  def sanitize_back_uri!
+    if params[:back_uri].present?
+      ok = request.protocol + request.host
+      render nothing: true, status: 403 if !params[:back_uri].start_with? ok
```

Comment:
Is `403` the desired behavior here? Just making sure that there aren't going to be unanticipated side effects.